Block registration: normalize blockType.parent to `array` (#66250)
WordPress/gutenberg: packages/blocks/CHANGELOG.md

## Unreleased

### Bug fixes

> - Enforce `blockType.parent` to be an array. While `parent` being a string wasn't never supported, it worked with some unintended side-effects that have been fixed by [#66250](https://github.com/WordPress/gutenberg/pull/66250). For that reason, we've added some code that automatically migrates strings to arrays — though it still raises a warning.

Reviewer: ```suggestion
- Enforce `blockType.parent` to be an array. While string values were never supported, they appeared to work with some unintended side-effects that have been fixed by [#66250](https://github.com/WordPress/gutenberg/pull/66250). For that reason, we've added some code that automatically migrates strings to arrays — though it still raises a warning.
```

Other notes:
* **enforce:** I'd either go with "cast" if we decide to do it automatically with no warnings, or "require" if we add warnings; enforce is more ambiguous to me.

## Unreleased

> - Enforce `blockType.parent` to be an array. While `parent` being a string wasn't never supported, it worked with some unintended side-effects that have been fixed by [#66250](https://github.com/WordPress/gutenberg/pull/66250). For that reason, we've added some code that automatically migrates strings to arrays — though it still raises a warning.

Author: Added a changelog to make sure this doesn't go unnoticed.

Reviewer 2: We should also add a heading to the change log entry, but I'm not sure what that should be.. Maybe `code quality`? 🤔

Author: Added `Bug fixes` heading https://github.com/WordPress/gutenberg/pull/66250/commits/0c4a9f55ec3e9f081274255691b27a79757c939d
